keras_contrib/applications/resnet.py

```diff
@@ -338,15 +481,19 @@ def ResNet(input_shape=None, classes=10, block='bottleneck', residual_unit='v2',
         raise ValueError('sigmoid activation can only be used when classes = 1')
     if repetitions is None:
         repetitions = [3, 4, 6, 3]
+
     # Determine proper input shape
-    input_shape = _obtain_input_shape(input_shape,
-                                      default_size=32,
-                                      min_size=8,
-                                      data_format=K.image_data_format(),
-                                      require_flatten=include_top)
-    _handle_dim_ordering()
-    if len(input_shape) != 3:
-        raise Exception("Input shape should be a tuple (nb_channels, nb_rows, nb_cols)")
+    input_shape = obtain_input_shape(input_shape=input_shape, require_flatten=include_top,
+                                     time_distributed=time_distributed)
+
+    _handle_dim_ordering(time_distributed=time_distributed, verbose=verbose)
+
+    if time_distributed:
+        if len(input_shape) != 4:
+            raise Exception("Input shape should be a tuple (nb_time_steps, nb_rows, nb_cols, nb_channels)")
```

Comment:
Since we are leaving out permutation of dims, maybe we should print an image_data_format-specific message here?

```
if time_distributed:
    if len(input_shape) != 4:
        if K.image_data_format() == 'channels_last':
            raise Exception("channels_last: Input shape should be a tuple (nb_time_steps, nb_rows, nb_cols, nb_channels)")
        else:
            raise Exception("channels_first: Input shape should be a tuple (nb_time_steps, nb_channels, nb_rows, nb_cols)")
else:
    if len(input_shape) != 3:
        if K.image_data_format() == 'channels_last':
            raise Exception("channels_last: Input shape should be a tuple (nb_rows, nb_cols, nb_channels)")
        else:
            raise Exception("channels_first: Input shape should be a tuple (nb_channels, nb_rows, nb_cols)")


```